src/components/Molecules/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,126 @@
+        const txnHashRegex = /^0x[a-fA-F0-9]{64}$/;
+        const blockRegex = /^\d+$/;
+        // const tokenRegex = /^[a-zA-Z0-9]+$/;
+        const domainNameRegex = /^[a-zA-Z0-9.-]+$/;
+
+        setSearchType("none");
+
+        if (addressRegex.test(searchInput)) {
+            setSearchType("address");
+        } else if (txnHashRegex.test(searchInput)) {
+            setSearchType("tx");
+        } else if (blockRegex.test(searchInput)) {
+            setSearchType("block");
+        }
+        // else if (tokenRegex.test(searchInput)) {
+        //     setSearchType("token");
+        // }
+        else if (domainNameRegex.test(searchInput)) {
+            setSearchType("address");
+        } else {
+            setSearchType("not found");
+        }
+    }, [searchInput]);
+
+    const handleResults = useCallback(() => {
+        switch (searchType) {
+            case "address": {
+                return (
+                    <div className="flex w-full flex-col gap-y-8">
+                        <AddressDetailsView
+                            address={searchInput}
+                            chain_name={selectedChain?.name as Chain}
+                            show_chain_selector={false}
+                        />
+                        <AddressActivityListView address={searchInput} />
+                    </div>
+                );
+            }
+            case "tx": {
+                return (
+                    <TransactionReceiptView
+                        chain_name={selectedChain?.name as Chain}
+                        tx_hash={searchInput}
+                    />
+                );
+            }
+            // case "token": {
+            //     return <>not sure how to handle this</>;
+            // }
+            case "block": {
+                return (
+                    <BlockDetails
+                        chain_name={selectedChain?.name as Chain}
+                        height={+searchInput}
+                    />
+                );
+            }
+            case "none": {
+                return <></>;
+            }
+            case "not found":
+            default: {
+                return <TypographyH3>{searchInput} not found</TypographyH3>;
+            }
+        }
+    }, [searchType, searchInput, selectedChain]);
+
+    return (
+        <main>
+            <div className="flex items-center gap-x-4">
+                <ChainSelector />
+                <Input
+                    type="text"
+                    name="search"
+                    value={searchInput}
+                    placeholder="Search by any Address / Txn Hash / Block / Domain Name"
+                    onChange={({ target: { value } }) => setSearchInput(value)}
+                    className="!border-accent-foreground"
+                />
+            </div>
+
+            {!searchInput ? (
+                <></>
+            ) : (
+                <div className="mt-4">{handleResults()}</div>
+            )}
```

Comment:
maybe add a button here to submit request.

Reply:
The search submission is automatic - use a debounce logic of 500ms, is the button still required?

Reply:
Not needed if we do what was suggested above with the helper function